xcollection/main.py

```diff
@@ -119,3 +119,20 @@ def _select_vars(dset):
             result = toolz.valfilter(_select_vars, self.datasets)
 
         return type(self)(datasets=result)
+
+    def map(self, func: typing.Callable[[xr.Dataset], xr.Dataset]) -> 'Collection':
+        """Apply a function to each dataset in the collection.
+        Parameters
+        ----------
+        func : callable
+            The function to apply to each dataset.
+
+        Returns
+        -------
+        Collection
+            A new collection containing the results of the function.
+
+        """
+        if not isinstance(func, typing.Callable):
+            raise TypeError(f'Expected a callable, got {type(func)}')
+        return type(self)(datasets=toolz.valmap(func, self.datasets))
```

Comment:
Should we relax this requirement i.e. allow `func` to return any type other than `xr.Dataset`, and return a dictionary of arbitrary values instead of a `Collection` object?

What do folks think?

Reply:
I think this is fine. The function can return DataArrays that will get promoted.

Reply:
If the user wants to return something random they could quite easily do

```
{k: func(ds) for k, ds in Collection.items()}
```

Reply:
Awesome! Sounds good